src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -407,6 +408,67 @@ uct_cuda_ipc_mem_dereg(uct_md_h md, const uct_md_mem_dereg_params_t *params)
     return UCS_OK;
 }
 
+static int
+uct_cuda_ipc_md_check_fabric_info(uct_cuda_ipc_md_t *md,
+                                  ucs_ternary_auto_value_t mnnvl_enable)
+{
+    static int mnnvl_supported = -1;
+#if HAVE_NVML_FABRIC_INFO
+    nvmlGpuFabricInfoV_t fabric_info;
+    nvmlDevice_t device;
+    ucs_status_t status;
+    char buf[64];
+
+    if (mnnvl_supported != -1) {
+        goto out;
+    }
+
+    if ((mnnvl_enable == UCS_NO) ||
+        (UCT_NVML_FUNC(nvmlInit_v2(), UCS_LOG_LEVEL_DIAG) != UCS_OK)) {
+        mnnvl_supported = 0;
+        goto out;
+    }
+
+    status = UCT_NVML_FUNC_LOG_ERR(nvmlDeviceGetHandleByIndex(0, &device));
+    if (status != UCS_OK) {
+        goto out_not_supported;
+    }
+
+    fabric_info.version = nvmlGpuFabricInfo_v2;
+    status              = UCT_NVML_FUNC_LOG_ERR(
+                             nvmlDeviceGetGpuFabricInfoV(device, &fabric_info));
+    if (status != UCS_OK) {
+        goto out_not_supported;
+    }
+
+    ucs_debug("fabric_info: healthmask=%u state=%u status=%u clique=%u uuid=%s",
+              fabric_info.healthMask, fabric_info.state, fabric_info.status,
+              fabric_info.cliqueId,
+              ucs_str_dump_hex(
+                  fabric_info.clusterUuid, NVML_GPU_FABRIC_UUID_LEN, buf,
+                  sizeof(buf), SIZE_MAX));
+
+    if ((fabric_info.state == NVML_GPU_FABRIC_STATE_COMPLETED) &&
+        (fabric_info.status == NVML_SUCCESS)) {
+        mnnvl_supported = 1;
+        goto out_sd;
+    }
+
+out_not_supported:
+    mnnvl_supported = 0;
+out_sd:
+    UCT_NVML_FUNC_LOG_ERR(nvmlShutdown());
+out:
+#else
+    mnnvl_supported = 0;
```

Comment:
maybe initialize mnnvl_supported to 0 instead of -1 when HAVE_NVML_FABRIC_INFO is not defined?